src/payment/models/crypto-input/crypto-input.service.ts

```diff
@@ -253,24 +259,59 @@ export class CryptoInputService {
     });
   }
 
-  private async saveAndForward(input: CryptoInput): Promise<void> {
+  private async testCompositeSwaps(
+    asset: string,
+    amount: number,
+    allowRetry = true,
+  ): Promise<{ btcAmount: number; usdtAmount: number }> {
     try {
-      // save
-      await this.cryptoInputRepo.save(input);
-      if (input.amlCheck === AmlCheck.FAIL) return;
+      const btcAmount = await this.client.testCompositeSwap(asset, 'BTC', amount);
+      const usdtAmount = await this.client.testCompositeSwap(asset, 'USDT', amount);
 
-      if (input.route.type === RouteType.STAKING) {
-        await this.cryptoStakingService.create(input);
+      return { btcAmount, usdtAmount };
+    } catch (e) {
+      try {
+        // poll the node
+        await this.client.getInfo();
+      } catch (nodeError) {
+        throw new NodeNotAccessibleError(NodeType.INPUT, nodeError);
       }
 
-      // forward (only await UTXO)
-      const targetAddress =
-        input.route.type === RouteType.SELL ? Config.node.dexWalletAddress : Config.node.stakingWalletAddress;
-      input.asset.type === AssetType.COIN
-        ? await this.forwardUtxo(input, targetAddress)
-        : await this.forwardToken(input, targetAddress);
-    } catch (e) {
-      console.error(`Failed to process crypto input ${input.id}:`, e);
+      if (allowRetry) {
+        // try once again
+        console.log('Retrying testCompositeSwaps after node poll success');
+        return await this.testCompositeSwaps(asset, amount, false);
+      }
+
+      // re-throw error, likely input related
+      throw e;
+    }
+  }
+
+  private async forwardInputs(): Promise<void> {
+    const inputs = await this.cryptoInputRepo.find({
+      where: { outTxId: '', amlCheck: AmlCheck.PASS },
+      relations: ['route'],
+    });
+
+    console.log(`Forwarding inputs (${inputs.length})`);
+
+    for (const input of inputs) {
+      try {
+        if (input.route.type === RouteType.STAKING) {
+          await this.cryptoStakingService.create(input);
+        }
+
+        // forward (only await UTXO)
```

Comment:
`(only await UTXO)` is an old comment, can be removed :-)

Reply:
done